pkg/common/common.go

```diff
@@ -92,3 +120,50 @@ func GenerateFileFromTemplate(templatedFilePath string, templateArgs interface{}
 
 	return generatedFilePath, nil
 }
+
+func IsRetriable(err error) bool {
+	for _, msg := range retriableErrors {
+		if strings.Contains(err.Error(), msg) {
+			return true
+		}
+	}
+	return false
+}
+
+func RetryOnError(backoff *wait.Backoff, retryExpected func(error) bool, fn condFunc) (interface{}, error) {
+	var ex, lastErr error
+	var out interface{}
+	caller := runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+	err := wait.ExponentialBackoff(*backoff, func() (bool, error) {
+		out, ex = fn()
+		switch {
+		case ex == nil:
+			return true, nil
+		case retryExpected(ex):
+			lastErr = ex
+			log.Warnf("A caller %v retried due to exception: %v", caller, ex)
+			return false, nil
+		default:
+			return false, ex
+		}
+	})
+	if err == wait.ErrWaitTimeout {
+		err = lastErr
+	}
+	return out, err
+}
+
+func GetEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+func GetUsernamePrefix() string {
```

Comment:
`IsRetriable`, `RetryOnError`, `GetEnv`, `GetUsernamePrefix` seem to be generic helper functions, they can go in the `util.go` file.

Reply:
done